itests/manual_cc_test.go

```diff
@@ -0,0 +1,216 @@
+	"github.com/filecoin-project/lotus/itests/kit"
+)
+
+// Manually onboard CC sectors, bypassing lotus-miner onboarding pathways
+
+func TestManualCCOnboarding(t *testing.T) {
+	req := require.New(t)
+	// kit.Expensive(t)
+
+	kit.QuietMiningLogs()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var (
+		//provingPeriod abi.ChainEpoch = 2880
+		blocktime = 2 * time.Millisecond
+
+		client kit.TestFullNode
+		minerA kit.TestMiner // A has some genesis sector, just keeps power
+		minerB kit.TestMiner // B pledges some sector
+	)
+
+	opts := []kit.NodeOpt{kit.WithAllSubsystems()}
+	ens := kit.NewEnsemble(t, kit.MockProofs()).
+		FullNode(&client, opts...).
+		Miner(&minerA, &client, opts...).
+		Start().
+		InterconnectAll()
+	ens.BeginMining(blocktime)
+
+	opts = append(opts, kit.OwnerAddr(client.DefaultKey))
+	ens.Miner(&minerB, &client, opts...).Start()
+
+	maddrA, err := minerA.ActorAddress(ctx)
+	req.NoError(err)
+
+	build.Clock.Sleep(time.Second)
+
+	t.Log("Submitting PreCommitSector...")
+
+	maddrB, err := minerB.ActorAddress(ctx)
+	req.NoError(err)
+
+	head, err := client.ChainHead(ctx)
+	req.NoError(err)
+
+	minerBInfo, err := client.StateMinerInfo(ctx, maddrB, head.Key())
+	req.NoError(err)
+
+	preCommitParams := &miner.PreCommitSectorBatchParams2{
+		Sectors: []miner.SectorPreCommitInfo{
+			{
+				Expiration:   2880 * 300,
+				SectorNumber: 22,
+				SealProof:    kit.TestSpt,
+
+				SealedCID:     cid.MustParse("bagboea4b5abcatlxechwbp7kjpjguna6r6q7ejrhe6mdp3lf34pmswn27pkkiekz"),
+				SealRandEpoch: head.Height() - 200,
+			},
+		},
+	}
+
+	enc := new(bytes.Buffer)
+	req.NoError(preCommitParams.MarshalCBOR(enc))
+
+	m, err := client.MpoolPushMessage(ctx, &types.Message{
+		To:     maddrB,
+		From:   minerB.OwnerKey.Address,
+		Value:  types.FromFil(1),
+		Method: builtin.MethodsMiner.PreCommitSectorBatch2,
+		Params: enc.Bytes(),
+	}, nil)
+	req.NoError(err)
+
+	r, err := client.StateWaitMsg(ctx, m.Cid(), 2, api.LookbackNoLimit, true)
+	req.NoError(err)
+	require.True(t, r.Receipt.ExitCode.IsSuccess())
+
+	client.WaitTillChain(ctx, kit.HeightAtLeast(r.Height+miner14.PreCommitChallengeDelay+5))
+
+	t.Log("Checking initial power...")
+
+	p, err := client.StateMinerPower(ctx, maddrA, r.TipSet)
+	req.NoError(err)
+	t.Logf("MinerA RBP: %v, QaP: %v", p.MinerPower.QualityAdjPower.String(), p.MinerPower.RawBytePower.String())
+
+	p, err = client.StateMinerPower(ctx, maddrB, r.TipSet)
+	req.NoError(err)
+	t.Logf("MinerB RBP: %v, QaP: %v", p.MinerPower.QualityAdjPower.String(), p.MinerPower.RawBytePower.String())
+	require.True(t, p.MinerPower.RawBytePower.IsZero())
+
+	t.Log("Submitting ProveCommitSector...")
+
+	bSectorNum := preCommitParams.Sectors[0].SectorNumber
+
+	proveCommitParams := miner14.ProveCommitSectors3Params{
+		SectorActivations:        []miner14.SectorActivationManifest{{SectorNumber: bSectorNum}},
+		SectorProofs:             [][]byte{{0xde, 0xad, 0xbe, 0xef}},
```

Comment:
Where did you get these values from ?

Reply:
They could be anything for now, this is using mock proofs which have a "always OK" effect when calling down in to the lower-level proofs (FFI) functions. Next step here would be to replace the mock proofs with real ones and do the FFI calls manually as well. We'll have to consider how much that buys us for NI-PoRep and whether it's worth the hassle. It would be good to do at some point though.

Reply:
If I understand correctly, I think it would be really nice to replace the mock proofs with real ones, as I suspect there is a lot of edge case testing that will be hard to do if the proofs always have a "always OK" effect, like: "Onboard a sectors, submit a fake/bad proofs, confirm that the disputer catches it"

Reply:
yeah, I'm in the process of figuring out how on earth to do that, it's buried under layers of abstraction that I'm trying to grok

Reply:
@rjan90 But looks like our existing itests use only mock proofs.

Reply:
So we've never done this kind of integration testing before.